hr_contract_cae/models/hr_contract.py

```diff
@@ -1,13 +1,109 @@
+import logging
+
 from dateutil.relativedelta import relativedelta
 
-from odoo import api, fields, models
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class ContractGroup(models.Model):
+    _name = "hr.contract.group"
+    _description = "Contract Group"
+    _order = "amendment_index"
+
+    contract_ids = fields.One2many(
+        comodel_name="hr.contract",
+        inverse_name="contract_group_id",
+        string="Contracts",
+    )
+
+    def get_initial(self):
+        return self.contract_ids.search(
+            [("contract_group_id", "=", self.id)], limit=1
+        )
```

Comment:
I'd do a computed field and (maybe) store the result but a function is fine too.